scalableminds/webknossos
# Dataset Upload: Improve heuristic to detect mag/layer-only dir

When users upload a wkw dataset with only a mag directory or only a layer directory (and no datasource-properties.json), there is a heuristic in place to fix the directory structure (i.e. create the mag parent or layer parent dir). Based on this, the explore route can later infer a datasource-properties.json.
However, the heuristic had false positives on numerical-only layer names. This PR replaces the heuristic with a new one that searches for "header.wkw" files and then looks at how deep in the directory tree they occur.

### Steps to test:
- Upload a normal wkw dataset, should still work
- Upload just the content of a layer directory, should be wrapped in "color" directory (see binaryData dir)
- Upload just the content of a mag directory, should be wrapped in "color/1" directory (see binaryData dir)
- Note that those repaired datasets will not be imported immediately. Rather, you need to go to the settings page first and import, to make them usable (i.e. create a datasource-properties.json file. This did not change in this PR)

### Issues:
- fixes #5884 

------
